src/routes/libraries.js

```diff
@@ -17,10 +19,22 @@ const maxQueryLength = 512;
  * @return {Promise<Object[]>}
  */
 const browse = async (query, searchFields) => {
+    // Normalize the search fields
+    const fields = searchFields.filter(field => validSearchFields.includes(field)).sort();
+
+    // Check if we have a cached result for this query
+    const cacheKey = await crypto.subtle.digest(
+        { name: 'SHA-512' },
+        new TextEncoder().encode(`${query}:${fields.join(',')}`),
+    ).then(buf => `libraries:${Array.from(new Uint8Array(buf)).map(b => b.toString(16).padStart(2, '0')).join('')}`);
```

Comment:
The code here isn't complex per se, but it is a bit difficult to read because there's so much going on in just one line. 

Would there be a way to make this a bit easier to read?

Reply:
Moved the `Array.from...` out to a dedicated `bufToHex` method to make it clearer :)